src/main/java/com/redhat/foreman/ForemanAPI.java

```diff
@@ -103,7 +103,12 @@ public JsonNode reserveHost(String hostname) {
      * @return string to be used for reserving.
      */
     private String getReserveReason() {
-        return "Reserved for " + Jenkins.getInstance().getRootUrl();
+        String url = "unknown";
+        Jenkins instance = Jenkins.getInstance();
+        if (instance != null) {
+            url = instance.getRootUrl();
```

Comment:
You are right, however plug-in can't get it in a common way... please read this thread https://github.com/jenkinsci/jenkins/pull/2297 where it is stated that current behavior is temporal only and will be removed soon (so final state will be `@Nonnull` annotation added to `getInstance()`).
Usage of `getActiveInstance()` is even worse because it is deprecated and it should never be born.
If someone disables throwing ISE via property, I don't thing it is a bad idea to throw NPE here. (the same behavior is in other places in the core, for ex. CLI extracted from the core what I've been reworking recently and community agrees with this approach)